### PR TITLE
Switch from CircleCI to GHA

### PR DESCRIPTION
**Pull request by FurqanHabibi**

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  PYTHON_VERSION: 3.8
```
Comment (ojii) [inline .github/workflows/workflow.yml:18]: could we make a build matrix to test python 3.7 ~ 3.11?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - run: curl -sSL https://install.python-poetry.org | python3 -
+      - uses: actions/cache@v3
+        with:
+          path: poetry.lock
+          key: ${{ github.sha }}
+      - run: poetry install
+      - uses: pre-commit/actions@v3.0.0
```
Comment (ojii) [inline .github/workflows/workflow.yml:69]: this replaces black/isort/mypy?

**Comment by ojii** [comment]
Since I committed the test fix to this branch I can no longer review this via github, so: I approve of these changes, and thank you!

**Comment by FurqanHabibi** [comment]
You should be able to approve it now!
No worries ✌️

**Review by ojii — APPROVED** [review]

**Pull request merged**